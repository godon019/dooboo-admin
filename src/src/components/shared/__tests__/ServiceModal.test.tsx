// import {
//   RenderResult,
//   act,
//   fireEvent,
//   render,
//   waitForDomChange,
//   waitForElement,
// } from '@testing-library/react';
// import ServiceModal, { ServiceModalProps } from '../ServiceModal';
// import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

// import React from 'react';
// import { ServiceFront } from '../../../types';
// import ThemeProvider from '../../../providers/ThemeProvider';
// import { getString } from '../../../../STRINGS';

// let component: React.ReactElement;
// let testingLib: any;
// const serviceMock: ServiceFront = {
//   id: '87355',
//   keyName: 'KEYNAME',
//   name: 'Davis, Howell and Lindgren',
//   nameKr: '국문이름',
//   thumbnail:
//     'https://s3.amazonaws.com/uifaces/faces/twitter/primozcigler/128.jpg',
//   image: 'https://s3.amazonaws.com/uifaces/faces/twitter/primozcigler/128.jpg',
//   homepage: 'http://whats.sub',
//   memo: 'whatssub',
//   iosAppStoreUrl: '',
//   androidPlayStoreUrl: '',
//   categoryId: '',
//   serviceGroupId: '',
// };
// // let initProps: Service = {
// //   thumbnail: null,
// //   image: null,
// //   name: '',
// //   homepage: '',
// //   memo: '',
// // };
// const SUPPORTED_FORMATS = ['image/gif', 'image/png'];
// const imageChangeValidation = (file) =>
//   file && SUPPORTED_FORMATS.includes(file.type);

// // const serviceInfo: Service = {
// //   id: '',
// //   name: initProps.name,
// //   thumbnail: initProps.thumbnail,
// // };

// let props: ServiceModalProps = {
//   onCloseServiceClick: jest.fn(),
//   onSubmitFinishedSuccess: jest.fn((arg) => arg),
//   onHandleImgChange: jest.fn(),
//   show: false,
//   serviceInfo: serviceMock,
// };

// describe('[ServiceModal] render', () => {
//   it('renders without crashing', () => {
//     const rendered: ReactTestRendererJSON = renderer
//       .create(
//         <ThemeProvider>
//           <ServiceModal {...props} />
//         </ThemeProvider>,
//       )
//       .toJSON();
//     expect(rendered).toMatchSnapshot();
//     expect(rendered).toBeTruthy();
//   });
// });

// // describe('[ServiceModal] interactions', () => {
// //   // const component = <ServiceModal {...props} />;
// //   const component = (
// //     <ThemeProvider>
// //       <ServiceModal {...props} />
// //     </ThemeProvider>
// //   );
// //   let renderResult: RenderResult;

// //   beforeAll(() => {
// //     renderResult = render(component);
// //   });
// //   it('modal closed inline method triggered', () => {
// //     const modalCloseButton = renderResult.getByTestId('serviceModalCloseBtn');
// //     fireEvent.click(modalCloseButton);
// //     expect(props.onCloseServiceClick).toHaveBeenCalled();
// //   });
// //   it('file type check when user upload thumbnail image file', async () => {
// //     let thumbnail = 'uploadThumbnail';
// //     let thumbnailInput = renderResult.getByTestId(`${thumbnail}-file`);
// //     const invalidFile = new File(['invalidfile'], 'invalidfile.pdf', {
// //       type: 'application/pdf',
// //     });
// //     fireEvent.change(thumbnailInput, { target: { files: [invalidFile] } });
// //     const defaultPreview = await waitForElement(() =>
// //       renderResult.getByTestId('defaultPreview'),
// //     );
// //     expect(defaultPreview).toBeTruthy();

// //     const validFile = new File(['validfile'], 'validfile.png', {
// //       type: 'image/png',
// //     });
// //     fireEvent.change(thumbnailInput, { target: { files: [validFile] } });
// //     const imageAreaAfterChange = await waitForElement(() =>
// //       renderResult.getByTestId(`${thumbnail}-preview`),
// //     );
// //     expect(imageAreaAfterChange).toBeTruthy();
// //   });
// //   it('existed data is rendered well on modal', () => {
// //     let newProps = {
// //       ...props,
// //       serviceInfo: serviceMock,
// //     };
// //     renderResult.rerender(
// //       <ThemeProvider>
// //         <ServiceModal {...newProps} />
// //       </ThemeProvider>,
// //     );

// //     const serviceNameInput = renderResult.getByTestId(
// //       'keyName',
// //     ) as HTMLInputElement;
// //     const serviceNameKorInput = renderResult.getByTestId(
// //       'serviceNameKor',
// //     ) as HTMLInputElement;
// //     const webPageInput = renderResult.getByTestId('homepage') as HTMLInputElement;
// //     const memoInput = renderResult.getByTestId('memo') as HTMLInputElement;

// //     expect(serviceNameInput.value).toBe(serviceMock.name);
// //     expect(serviceNameKorInput.value).toBe(serviceMock.nameKr);
// //     expect(webPageInput.value).toBe(serviceMock.homepage);
// //     expect(memoInput.value).toBe(serviceMock.memo);
// //   });
// //   it('onSubmit inline method triggered', async () => {
// //     // TODO: get to be triggered
// //     const modalSaveButton = renderResult.getByTestId('serviceSubmit');
// //     fireEvent.submit(modalSaveButton);
// //     setTimeout(() => {
// //       expect(props.onSubmitClick).toHaveBeenCalled();
// //     }, 410);
// //   });
// // });
