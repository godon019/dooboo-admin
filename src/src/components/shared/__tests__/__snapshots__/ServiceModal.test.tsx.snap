// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ServiceModal] render renders without crashing 1`] = `
<div
  className="sc-dxgOiQ ksLhkK"
>
  <div
    className="sc-ckVGcZ bPGTGF"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="sc-kAzzGY jQociM"
      >
        <div
          className="sc-chPdSV kYAPgs"
        >
          <img
            className="sc-kGXeez imfAWV"
            src="https://github.com/whatssub/whatssub-mobile/blob/master/assets/splash.gif?raw=true"
          />
          <span
            className="sc-kgoBCf lhwMoN"
          >
            | Admin Page
          </span>
        </div>
        <div
          className="sc-kpOJdX dyVmsv"
        >
          + 서비스 추가
        </div>
      </div>
      <div
        className="sc-jKJlTe cHRoYJ"
      >
        <div
          className="sc-iAyFgw ieLOwj"
        >
          <div
            className="sc-hSdWYo ibQGur"
          >
            키값
          </div>
          <div
            className="sc-eHgmQL irHKpW"
          >
            <div
              className="sc-jTzLTM czpJAl"
            >
              <input
                className="sc-fjdhpX dqwVOL"
                data-testid="keyName"
                id="keyName"
                name="keyName"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="KEYNAME"
              />
            </div>
          </div>
        </div>
        <div
          className="sc-iAyFgw ieLOwj"
        >
          <div
            className="sc-hSdWYo ibQGur"
          >
            썸네일
          </div>
          <div
            className="sc-eHgmQL irHKpW"
          >
            <div
              className="sc-bZQynM jwgxKa"
            >
              <div
                className="sc-htoDjs iKOCRY"
              >
                <img
                  className="sc-gqjmRU dmhDdt"
                  data-testid="thumbnail-preview"
                  src="https://s3.amazonaws.com/uifaces/faces/twitter/primozcigler/128.jpg"
                />
                <input
                  className="sc-gzVnrw aZFdW"
                  data-testid="thumbnail-file"
                  id="thumbnail"
                  name="thumbnail"
                  onChange={[Function]}
                  type="file"
                />
              </div>
              <div
                className="sc-dnqmqq kHYyxp"
              >
                <button
                  className="sc-bdVaJa bEdxjs"
                  data-testid="thumbnail-upload-btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "borderColor": "#069ccd",
                      "height": null,
                      "marginBottom": 0,
                      "padding": ".7rem",
                    }
                  }
                  type="button"
                >
                  <div>
                    <span
                      className="sc-bxivhb koiFyU"
                      style={
                        Object {
                          "color": "#069ccd",
                        }
                      }
                    >
                      이미지 첨부
                    </span>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-iAyFgw ieLOwj"
        >
          <div
            className="sc-hSdWYo ibQGur"
          >
            이미지
          </div>
          <div
            className="sc-eHgmQL irHKpW"
          >
            <div
              className="sc-bZQynM jwgxKa"
            >
              <div
                className="sc-htoDjs iKOCRY"
              >
                <img
                  className="sc-gqjmRU dmhDdt"
                  data-testid="image-preview"
                  src="https://s3.amazonaws.com/uifaces/faces/twitter/primozcigler/128.jpg"
                />
                <input
                  className="sc-gzVnrw aZFdW"
                  data-testid="image-file"
                  id="image"
                  name="image"
                  onChange={[Function]}
                  type="file"
                />
              </div>
              <div
                className="sc-dnqmqq kHYyxp"
              >
                <button
                  className="sc-bdVaJa bEdxjs"
                  data-testid="image-upload-btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "borderColor": "#069ccd",
                      "height": null,
                      "marginBottom": 0,
                      "padding": ".7rem",
                    }
                  }
                  type="button"
                >
                  <div>
                    <span
                      className="sc-bxivhb koiFyU"
                      style={
                        Object {
                          "color": "#069ccd",
                        }
                      }
                    >
                      이미지 첨부
                    </span>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-iAyFgw ieLOwj"
        >
          <div
            className="sc-hSdWYo ibQGur"
          >
            Web url
          </div>
          <div
            className="sc-eHgmQL irHKpW"
          >
            <div
              className="sc-jTzLTM czpJAl"
            >
              <input
                className="sc-fjdhpX dqwVOL"
                data-testid="homepage"
                id="homepage"
                name="homepage"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="http://whats.sub"
              />
            </div>
          </div>
        </div>
        <div
          className="sc-iAyFgw ieLOwj"
        >
          <div
            className="sc-hSdWYo ibQGur"
          >
            안드로이드 url
          </div>
          <div
            className="sc-eHgmQL irHKpW"
          >
            <div
              className="sc-jTzLTM czpJAl"
            >
              <input
                className="sc-fjdhpX dqwVOL"
                data-testid="androidPlayStoreUrl"
                id="androidPlayStoreUrl"
                name="androidPlayStoreUrl"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="http://whats.sub"
              />
            </div>
          </div>
        </div>
        <div
          className="sc-iAyFgw ieLOwj"
        >
          <div
            className="sc-hSdWYo ibQGur"
          >
            IOS url
          </div>
          <div
            className="sc-eHgmQL irHKpW"
          >
            <div
              className="sc-jTzLTM czpJAl"
            >
              <input
                className="sc-fjdhpX dqwVOL"
                data-testid="iosAppStoreUrl"
                id="iosAppStoreUrl"
                name="iosAppStoreUrl"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
                value="http://whats.sub"
              />
            </div>
          </div>
        </div>
        <div
          className="sc-iAyFgw ieLOwj"
        >
          <div
            className="sc-hSdWYo ibQGur"
          >
            부가 설명
          </div>
          <div
            className="sc-eHgmQL irHKpW"
          >
            <textarea
              className="sc-cSHVUG ikreCl"
              data-testid="memo"
              id="memo"
              name="memo"
              onBlur={[Function]}
              onChange={[Function]}
              rows={10}
              value="whatssub"
            />
          </div>
        </div>
      </div>
      <div
        className="sc-eNQAEJ xpVzC"
      >
        <button
          className="sc-bwzfXH fWdDwp"
          data-testid="serviceModalCloseBtn"
          onClick={[Function]}
          style={
            Object {
              "height": "60px",
              "width": "25%",
            }
          }
          type="button"
        >
          <div>
            <span
              className="sc-htpNat esLJJJ"
              style={Object {}}
            >
              취소
            </span>
          </div>
        </button>
        <button
          className="sc-bdVaJa bEdxjs"
          data-testid="serviceSubmit"
          style={
            Object {
              "height": "60px",
              "width": "25%",
            }
          }
          type="submit"
        >
          <div>
            <span
              className="sc-bxivhb koiFyU"
              style={Object {}}
            >
              등록
            </span>
          </div>
        </button>
      </div>
    </form>
  </div>
</div>
`;
